CI - Fix backup_task_after_failure race

## Problem

BDB `backups` failures recur in recent failed-main runs, with the representative failure in `backup_test.py::test_backup_task_after_failure`.

Recent signal from CI sweep:
- BDB `backups` failed in 6/20 sampled failed-main runs.
- Representative run/job: https://github.com/389ds/389-ds-base/actions/runs/26732183826, job `78778957716`.

Representative failure:

```text
dirsrvtests/tests/suites/backups/backup_test.py::test_backup_task_after_failure FAILED
E   OSError: [Errno 39] Directory not empty: '/var/lib/dirsrv/slapd-standalone1/bak/bak1' -> '/var/lib/dirsrv/slapd-standalone1/bak/bak1b'
dirsrvtests/tests/suites/backups/backup_test.py:345
```

The test tries to rename a backup directory while waiting for backup completion, which appears race-prone when the directory is actively being written.

## Proposed work

Investigate the intended failure injection for the backup task. Make the test force a deterministic backup failure without depending on an unsafe directory rename race, or adjust the backup code/test synchronization if the current behavior exposes a real bug.

## Acceptance criteria

- `backup_test.py::test_backup_task_after_failure` passes reliably in BDB CI.
- The test still verifies backup failure handling and a subsequent successful backup.
- The failure injection is deterministic and does not depend on timing of directory contents.

--- a/dirsrvtests/tests/suites/backups/backup_test.py
+++ b/dirsrvtests/tests/suites/backups/backup_test.py
@@ -327,7 +327,7 @@ def test_backup_task_after_failure(mytopo):
     inst = mytopo.standalone
     tasks = inst.tasks
     archive_dir1 = f'{inst.ds_paths.backup_dir}/bak1'
-    archive_dir1b = f'{inst.ds_paths.backup_dir}/bak1b'
+    dir1bidx = 1
     archive_dir2 = f'{inst.ds_paths.backup_dir}/bak2'
 
     # Sometime the backup complete too fast, so lets retry if first
@@ -342,6 +342,8 @@ def test_backup_task_after_failure(mytopo):
         done,exitCode,warningCode = (False, None, None)
         while not done:
             if os.path.isdir(archive_dir1):
+                archive_dir1b = f'{inst.ds_paths.backup_dir}/bak1b{dir1bidx}'
+                dir1bidx += 1
                 os.rename(archive_dir1, archive_dir1b)
             done,exitCode,warningCode = tasks.checkTask(tasks.entry)
             time.sleep(0.01)

